corehq/apps/app_manager/views/modules.py

```diff
@@ -359,15 +361,33 @@ def _setup_case_property_builder(app):
 
 # Parent case selection in case list: get modules whose case type is the parent of the given module's case type
 def _get_modules_with_parent_case_type(app, module, case_property_builder, case_type_):
-        parent_types = case_property_builder.get_parent_types(case_type_)
-        modules = app.modules
-        parent_module_ids = [mod.unique_id for mod in modules
-                             if mod.case_type in parent_types]
-        return [{
-            'unique_id': mod.unique_id,
-            'name': mod.name,
-            'is_parent': mod.unique_id in parent_module_ids,
-        } for mod in app.modules if mod.case_type != case_type_ and mod.unique_id != module.unique_id]
+
+    parent_types = case_property_builder.get_parent_types(case_type_)
+    modules = app.modules
+    parent_module_ids = [
+        mod.unique_id for mod in modules
+        if mod.case_type in parent_types]
+
+    return [{
+        'unique_id': mod.unique_id,
+        'name': mod.name,
+        'is_parent': mod.unique_id in parent_module_ids,
+    } for mod in app.modules if mod.case_type != case_type_ and mod.unique_id != module.unique_id]
+
+
+def _get_all_case_modules(app, module, case_property_builder, case_type_):
+    # return all modules except the given module
+    parent_types = case_property_builder.get_parent_types(case_type_)
+    modules = app.modules
+    parent_module_ids = [
+        mod.unique_id for mod in modules
+        if mod.case_type in parent_types]
+
+    return [{
+        'unique_id': mod.unique_id,
+        'name': mod.name,
+        'is_parent': mod.unique_id in parent_module_ids,
+    } for mod in app.modules if mod.unique_id != module.unique_id]
```

Comment:
Should this also make sure that the module has a non-null case_type attr?

Reply:
It should only return modules with a case type, yes. Looking at this again, I think you can drop the `parent_types` logic above. It looks like that's only used to star the modules where the case type is a parent of the current module, but for this new workflow, users aren't looking for a parent module so I don't think they need the parents starred.

Reply:
Ah, makes sense. So, should we modify this to include modules of the same case-type as the given module (excluding the given module)? What should we do if there were no other modules of that case-type?

Reply:
I think the friendliest option would be to display a message like "No menus available" in place of the dropdown, what do you think of that? We could also hide the option and dropdown if there aren't appropriate menus available. In general, I think showing disabled UIs, with an explanation for why they're disabled, is clearer than hiding them.

Out of curiosity, I checked what prod does if you check the "Select Parent First" box but there aren't any menus. You get a dropdown that says "Unknown Menu" and an error saying "This menu could not be found." It's good that it shows you there's a problem, but the message is a bit confusing.

Reply:
> You get a dropdown that says "Unknown Menu" and an error saying "This menu could not be found."

@orangejenny Yup, I found this so as well and found the behaviour satisfactory. Agreed that the language is a slightly confusing. That special dropdown gets displayed when a menu is not selected or not available. I have changed it to `Menu not set or not available` now.